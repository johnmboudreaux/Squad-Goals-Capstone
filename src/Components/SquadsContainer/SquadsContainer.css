.squads-body {
  background-color: #3782AD;
  box-shadow: 2px 2px 4px #282828;
  border-radius: 5px;
  color: #d2d3d3;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  margin: 10px auto 10px auto;
  opacity: .95;
  text-align: center;
  height: auto; }

.squads-body-title {
  font-weight: lighter;
  font-size: 1.5em;
  margin: 10px 0 15px 0; }

.squads-body-info tbody {
  width: 100%; }

.squads-body input {
  background-color: #57a2cd;
  border: none;
  border-radius: 3px;
  margin-bottom: 30px; }

.squads-body input:focus {
  outline: none; }

.squad-row a {
  color: #d2d3d3;
  display: flex;
  text-decoration: none; }

.squad-row th {
  font-size: 1.4em;
  font-weight: lighter; }
