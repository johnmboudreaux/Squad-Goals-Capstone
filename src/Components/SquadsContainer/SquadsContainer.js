import React from 'react';
import PropTypes from 'prop-types';
import SquadCard from '../SquadCard/SquadCard';

const SquadsContainer = ({
  squadData,
  createNewSquad,
  leaveGroup,
  userId,
}) => {
  const squadsInfo = squadData.map((squad) => {
    return (
      <SquadCard
        userId={userId}
        leaveGroup={leaveGroup}
        squad={squad}
        key={`squad-${squad.id}`}
      />
    );
  });

  return (
    <div className="squads-body">
      <div>
        <h2 className="squads-body-title">Squads</h2>
        <div className="squads-body-info">
          <div className="squad-row">
            <h3>Squad Title</h3>
            <h3>Competitors</h3>
            <h3>Squad Goals</h3>
          </div>
          { squadsInfo }
        </div>
      </div>
    </div>
  );
};

SquadsContainer.propTypes = {
  squadData: PropTypes.arrayOf(PropTypes.object),
  createNewSquad: PropTypes.func,
  leaveGroup: PropTypes.func,
  userId: PropTypes.number,
};

export default SquadsContainer;
