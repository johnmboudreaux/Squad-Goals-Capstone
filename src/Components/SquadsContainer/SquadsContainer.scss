@import '../../variables.scss';

.squads-body {
  background-color: $main;
  box-shadow: 2px 2px 4px $black;
  border-radius: 5px;
  color: $white;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 50vw;
  width: 50vw;
  margin: 10px auto 10px auto;
  opacity: .95;
  text-align: center;
}

.squads-body-title {
  font-weight: lighter;
  font-size: 1.5em;
  margin: 10px 0 15px 0;
}

.squads-body-info {
  font-size: .8em;
  margin: 10px;
}

.squads-body input {
  background-color: $main-light;
  border: none;
  border-radius: 3px;
  margin-bottom: 30px;
  width: 45%;
}

.squads-body input:focus {
  outline: none;
}
