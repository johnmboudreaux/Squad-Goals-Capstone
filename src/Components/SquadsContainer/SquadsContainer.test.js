import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import SquadsContainer from './SquadsContainer';

describe('SquadsContainer snapshot', () => {
  const squad = [{
    conversation_id: 15,
    created_at: '2018-01-08',
    id: 9,
    squad_name: 'El Squadro',
    updated_at: '2018-01-08T20:56:36.8072',
  }];

  it('should always match the snapshot', () => {
    const wrapper = shallow(<SquadsContainer
      squadData={squad}
    />);

    expect(wrapper).toMatchSnapshot();
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><SquadsContainer squadData={squad} /></BrowserRouter>, div);
  });
});
