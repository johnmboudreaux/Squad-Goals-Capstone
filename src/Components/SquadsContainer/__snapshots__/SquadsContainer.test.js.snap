// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquadsContainer snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SquadsContainer
    squadData={
      Array [
        Object {
          "conversation_id": 15,
          "created_at": "2018-01-08",
          "id": 9,
          "squad_name": "El Squadro",
          "updated_at": "2018-01-08T20:56:36.8072",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h2
          className="squads-body-title"
        >
          Squads
        </h2>
        <div
          className="squads-body-info"
        >
          <div
            className="squad-row"
          >
            <h3
              className="row-header"
            >
              Squad Title
            </h3>
            <h3
              className="row-header"
            >
              Competitors
            </h3>
            <h3
              className="row-header"
            >
              Squad Goals
            </h3>
          </div>
          <SquadCard
            leaveGroup={undefined}
            squad={
              Object {
                "conversation_id": 15,
                "created_at": "2018-01-08",
                "id": 9,
                "squad_name": "El Squadro",
                "updated_at": "2018-01-08T20:56:36.8072",
              }
            }
            userId={undefined}
          />
        </div>
      </div>,
      "className": "squads-body",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="squads-body-title"
          >
            Squads
          </h2>,
          <div
            className="squads-body-info"
          >
            <div
              className="squad-row"
            >
              <h3
                className="row-header"
              >
                Squad Title
              </h3>
              <h3
                className="row-header"
              >
                Competitors
              </h3>
              <h3
                className="row-header"
              >
                Squad Goals
              </h3>
            </div>
            <SquadCard
              leaveGroup={undefined}
              squad={
                Object {
                  "conversation_id": 15,
                  "created_at": "2018-01-08",
                  "id": 9,
                  "squad_name": "El Squadro",
                  "updated_at": "2018-01-08T20:56:36.8072",
                }
              }
              userId={undefined}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Squads",
            "className": "squads-body-title",
          },
          "ref": null,
          "rendered": "Squads",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="squad-row"
              >
                <h3
                  className="row-header"
                >
                  Squad Title
                </h3>
                <h3
                  className="row-header"
                >
                  Competitors
                </h3>
                <h3
                  className="row-header"
                >
                  Squad Goals
                </h3>
              </div>,
              Array [
                <SquadCard
                  leaveGroup={undefined}
                  squad={
                    Object {
                      "conversation_id": 15,
                      "created_at": "2018-01-08",
                      "id": 9,
                      "squad_name": "El Squadro",
                      "updated_at": "2018-01-08T20:56:36.8072",
                    }
                  }
                  userId={undefined}
                />,
              ],
            ],
            "className": "squads-body-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="row-header"
                  >
                    Squad Title
                  </h3>,
                  <h3
                    className="row-header"
                  >
                    Competitors
                  </h3>,
                  <h3
                    className="row-header"
                  >
                    Squad Goals
                  </h3>,
                ],
                "className": "squad-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Squad Title",
                    "className": "row-header",
                  },
                  "ref": null,
                  "rendered": "Squad Title",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Competitors",
                    "className": "row-header",
                  },
                  "ref": null,
                  "rendered": "Competitors",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Squad Goals",
                    "className": "row-header",
                  },
                  "ref": null,
                  "rendered": "Squad Goals",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "squad-9",
              "nodeType": "function",
              "props": Object {
                "leaveGroup": undefined,
                "squad": Object {
                  "conversation_id": 15,
                  "created_at": "2018-01-08",
                  "id": 9,
                  "squad_name": "El Squadro",
                  "updated_at": "2018-01-08T20:56:36.8072",
                },
                "userId": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h2
            className="squads-body-title"
          >
            Squads
          </h2>
          <div
            className="squads-body-info"
          >
            <div
              className="squad-row"
            >
              <h3
                className="row-header"
              >
                Squad Title
              </h3>
              <h3
                className="row-header"
              >
                Competitors
              </h3>
              <h3
                className="row-header"
              >
                Squad Goals
              </h3>
            </div>
            <SquadCard
              leaveGroup={undefined}
              squad={
                Object {
                  "conversation_id": 15,
                  "created_at": "2018-01-08",
                  "id": 9,
                  "squad_name": "El Squadro",
                  "updated_at": "2018-01-08T20:56:36.8072",
                }
              }
              userId={undefined}
            />
          </div>
        </div>,
        "className": "squads-body",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="squads-body-title"
            >
              Squads
            </h2>,
            <div
              className="squads-body-info"
            >
              <div
                className="squad-row"
              >
                <h3
                  className="row-header"
                >
                  Squad Title
                </h3>
                <h3
                  className="row-header"
                >
                  Competitors
                </h3>
                <h3
                  className="row-header"
                >
                  Squad Goals
                </h3>
              </div>
              <SquadCard
                leaveGroup={undefined}
                squad={
                  Object {
                    "conversation_id": 15,
                    "created_at": "2018-01-08",
                    "id": 9,
                    "squad_name": "El Squadro",
                    "updated_at": "2018-01-08T20:56:36.8072",
                  }
                }
                userId={undefined}
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Squads",
              "className": "squads-body-title",
            },
            "ref": null,
            "rendered": "Squads",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="squad-row"
                >
                  <h3
                    className="row-header"
                  >
                    Squad Title
                  </h3>
                  <h3
                    className="row-header"
                  >
                    Competitors
                  </h3>
                  <h3
                    className="row-header"
                  >
                    Squad Goals
                  </h3>
                </div>,
                Array [
                  <SquadCard
                    leaveGroup={undefined}
                    squad={
                      Object {
                        "conversation_id": 15,
                        "created_at": "2018-01-08",
                        "id": 9,
                        "squad_name": "El Squadro",
                        "updated_at": "2018-01-08T20:56:36.8072",
                      }
                    }
                    userId={undefined}
                  />,
                ],
              ],
              "className": "squads-body-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3
                      className="row-header"
                    >
                      Squad Title
                    </h3>,
                    <h3
                      className="row-header"
                    >
                      Competitors
                    </h3>,
                    <h3
                      className="row-header"
                    >
                      Squad Goals
                    </h3>,
                  ],
                  "className": "squad-row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Squad Title",
                      "className": "row-header",
                    },
                    "ref": null,
                    "rendered": "Squad Title",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Competitors",
                      "className": "row-header",
                    },
                    "ref": null,
                    "rendered": "Competitors",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Squad Goals",
                      "className": "row-header",
                    },
                    "ref": null,
                    "rendered": "Squad Goals",
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": "squad-9",
                "nodeType": "function",
                "props": Object {
                  "leaveGroup": undefined,
                  "squad": Object {
                    "conversation_id": 15,
                    "created_at": "2018-01-08",
                    "id": 9,
                    "squad_name": "El Squadro",
                    "updated_at": "2018-01-08T20:56:36.8072",
                  },
                  "userId": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
